azurelinuxagent/common/protocol/wire.py

```diff
@@ -1041,10 +1065,23 @@ def send_encoded_event(self, provider_id, event_str, encoding='utf8'):
         # dividing it into parts.
         data = data_format_header + event_str + data_format_footer
         try:
-            header = self.get_header_for_xml_content()
             # NOTE: The call to wireserver requests utf-8 encoding in the headers, but the body should not
             #       be encoded: some nodes in the telemetry pipeline do not support utf-8 encoding.
-            resp = self.call_wireserver(restutil.http_post, uri, data, header)
+            header = self.get_header_for_xml_content()
+
+            # wait until throttling limit reset to make next call
+            while not can_make_wireserver_call():
+                next_call_time = self.telemetry_api_calls_timestamps[0] + timedelta(seconds=TELEMETRY_INTERVAL)
+                logger.verbose("Reached telemetry api throttling limit: {0}, so waiting to make next call after : {1}".format(TELEMETRY_MAX_CALLS_PER_INTERVAL, next_call_time))
+                time.sleep(TELEMETRY_DELAY)
```

Comment:
this sleeps for 1s, shouldn't it continue sleep until 'next_call_time'?

Reply:
well it will sleep and check and eventually it will wait until next time. Initially total seconds computation not available in 2.6 to sleep until_next_time.  Now i found we have done manual computation in periodic operation, so using same formula